src/category_theory/sites/dense_subsite.lean

```diff
@@ -0,0 +1,416 @@
+      f.unop ≫ app_hom H (hom_over α (unop X)) (unop U) x,
+    symmetry,
+    apply sheaf_eq_amalgamation,
+    apply H.is_cover,
+    intros Y' f' hf',
+    change unop X ⟶ ℱ.obj (op (unop _)) at x,
+    simp only [pushforward_family, functor.comp_map,
+      coyoneda_obj_map, hom_over_app, category.assoc],
+    congr' 1,
+    conv_lhs { rw ← hf'.some.fac },
+    simp only [← category.assoc, op_comp, functor.map_comp],
+    congr' 1,
+    refine (app_hom_restrict H (hom_over α (unop X)) hf'.some.map.op x).trans _,
+    simp
+  end }
+
+/--
+(Implementation). `sheaf_coyoneda_hom` but the order of the arguments of the functor are swapped.
+-/
+noncomputable
+def sheaf_yoneda_hom (α : G.op ⋙ ℱ ⟶ G.op ⋙ ℱ'.val) :
+  ℱ ⋙ yoneda ⟶ ℱ'.val ⋙ yoneda :=
+begin
+  let α := sheaf_coyoneda_hom H α,
+  refine { app := _, naturality' := _ },
+  { intro U,
+    refine { app := λ X, (α.app X).app U,
+      naturality' := λ X Y f, by simpa using congr_app (α.naturality f) U } },
+  { intros U V i,
+    ext X x,
+    exact congr_fun ((α.app X).naturality i) x },
+end
+
+/--
+Given an natural transformation `G ⋙ ℱ ⟶ G ⋙ ℱ'` between presheaves of arbitrary category,
+where `G` is full and cover-dense, and `ℱ'` is a sheaf, we may obtain a natural transformation
+between presheaves.
+-/
+noncomputable
+def sheaf_hom (α : G.op ⋙ ℱ ⟶ G.op ⋙ ℱ'.val) :
+  ℱ ⟶ ℱ'.val :=
+let α' := sheaf_yoneda_hom H α in
+  { app := λ X, yoneda.preimage (α'.app X),
+    naturality' := λ X Y f, yoneda.map_injective (by simpa using α'.naturality f) }
+
+/--
+Given an natural isomorphism `G ⋙ ℱ ≅ G ⋙ ℱ'` between presheaves of arbitrary category,
+where `G` is full and cover-dense, and `ℱ', ℱ` are sheaves,
+we may obtain a natural isomorphism between presheaves.
+-/
+@[simps] noncomputable
+def presheaf_iso {ℱ ℱ' : Sheaf K A} (i : G.op ⋙ ℱ.val ≅ G.op ⋙ ℱ'.val) :
+  ℱ.val ≅ ℱ'.val :=
+begin
+  haveI : ∀ (X : Dᵒᵖ), is_iso ((sheaf_hom H i.hom).app X),
+  { intro X,
+    apply is_iso_of_reflects_iso _ yoneda,
+    use (sheaf_yoneda_hom H i.inv).app X,
+    split;
+      ext x : 2;
+      simp only [sheaf_hom, nat_trans.comp_app, nat_trans.id_app, functor.image_preimage],
+      exact ((presheaf_iso H (iso_over i (unop x))).app X).hom_inv_id,
+      exact ((presheaf_iso H (iso_over i (unop x))).app X).inv_hom_id,
+    apply_instance },
+  haveI : is_iso (sheaf_hom H i.hom) := by apply nat_iso.is_iso_of_is_iso_app,
+  apply as_iso (sheaf_hom H i.hom),
+end
+
+/--
+Given an natural isomorphism `G ⋙ ℱ ≅ G ⋙ ℱ'` between presheaves of arbitrary category,
+where `G` is full and cover-dense, and `ℱ', ℱ` are sheaves,
+we may obtain a natural isomorphism between presheaves.
+-/
+@[simps] noncomputable
+def sheaf_iso {ℱ ℱ' : Sheaf K A} (i : G.op ⋙ ℱ.val ≅ G.op ⋙ ℱ'.val) : ℱ ≅ ℱ' :=
+{ hom := (presheaf_iso H i).hom,
+  inv := (presheaf_iso H i).inv,
+  hom_inv_id' := (presheaf_iso H i).hom_inv_id,
+  inv_hom_id' := (presheaf_iso H i).inv_hom_id }
```

Comment:
This and [another sheaf_iso above](https://github.com/leanprover-community/mathlib/pull/9694/files#diff-4a971c16350870036568147a6226f23328a30c23e6a46a94279ca00224fefe35R245-R247) don't follow the same format. And can't this simply be `presheaf_iso H i` with some arguments made explicit? Or is it for the purpose of `simps` that you wrote down the individual fields? (I only noticed this after seeing a review message sent to my mailbox.)